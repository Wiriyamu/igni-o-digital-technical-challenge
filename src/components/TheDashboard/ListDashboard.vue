<template>
  <section class="list">
    <div class="list__actions">
      <p>200 resultados encontrados</p>

      <div class="list__buttons">
        <select class="list__select">
          <option>Listagem</option>
        </select>

        <figure class="list__grid">
          <img src="@/assets/images/grid.svg" alt="Grid" />
        </figure>

        <figure class="list__list">
          <img src="@/assets/images/list.svg" alt="Lista" />
        </figure>
      </div>
    </div>

    <input class="list__search" placeholder="Buscar" />

    <list-item-dashboard />
    <list-item-dashboard />
    <list-item-dashboard />
    <list-item-dashboard />
    <list-item-dashboard />
  </section>
</template>

<script>
import ListItemDashboard from './ListItemDashboard.vue'
export default {
  components: { ListItemDashboard },
  name: 'ListDashboard'
}
</script>

<style lang="scss" scoped>
.list {
  width: 100%;
  margin: 30px 20px 0 20px;

  &__actions {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__buttons {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__select {
    width: 130px;
    padding: 5px;
    border: 0;
    border-radius: 5px;
  }

  &__grid,
  &__list {
    display: flex;
    justify-content: center;
    width: 30px;
    height: 30px;
    margin-left: 10px;
    border-radius: 5px;
    background-color: #fff;
    cursor: pointer;

    & img {
      width: 15px;
    }
  }

  &__search {
    width: 100%;
    height: 50px;
    margin: 20px 0 30px 0;
    padding: 20px;
    border: 0;
    border-radius: 5px;
  }
}
</style>
