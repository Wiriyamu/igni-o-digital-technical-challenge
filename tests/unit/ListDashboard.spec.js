import ListDashboard from '@/components/TheDashboard/ListDashboard'
import { mount } from '@vue/test-utils'

describe('ListDashboard.vue', () => {
  it('Snapshot component', () => {
    const wrapper = mount(ListDashboard)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
