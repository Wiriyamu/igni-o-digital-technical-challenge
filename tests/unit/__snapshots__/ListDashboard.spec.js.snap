// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDashboard.vue Snapshot component 1`] = `
<section class="list">
  <div class="list__actions">
    <p class="list__amount">200 resultados encontrados</p>
    <div class="list__buttons"><select class="list__select">
        <option>Listagem</option>
      </select>
      <figure class="list__grid"><img src="" alt="Grid"></figure>
      <figure class="list__list"><img src="" alt="Lista"></figure>
    </div>
  </div>
  <div class="list__search-container"><input class="list__search" placeholder="Buscar"><img class="list__search-icon" src="" alt="Pesquisa"></div>
  <section class="item">
    <figure class="item__photo-container"><img class="item__photo" src="" alt="Photo"></figure>
    <div class="item__texts">
      <p class="item__title">Fórmula de lançamento</p>
      <p class="item__author">Por <strong>Érico Rocha</strong></p>
      <p class="item__description"> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. </p>
    </div>
    <div class="item__buy">
      <div class="item__star-button">
        <p class="item__star-note">3.5</p><img class="item__star" src="" alt="estrela">
      </div>
      <div class="item__buttons">
        <p class="item__price">R$ 2550</p>
        <div class="item__heart-box"><img src="" alt="Amei">
          <p class="item__heart-text">Amei</p>
        </div>
        <div class="item__bag-box"><img src="" alt="Comprar">
          <p class="item__bag-text">Comprar</p>
        </div>
      </div>
    </div>
  </section>
  <section class="item">
    <figure class="item__photo-container"><img class="item__photo" src="" alt="Photo"></figure>
    <div class="item__texts">
      <p class="item__title">Fórmula de lançamento</p>
      <p class="item__author">Por <strong>Érico Rocha</strong></p>
      <p class="item__description"> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. </p>
    </div>
    <div class="item__buy">
      <div class="item__star-button">
        <p class="item__star-note">3.5</p><img class="item__star" src="" alt="estrela">
      </div>
      <div class="item__buttons">
        <p class="item__price">R$ 2550</p>
        <div class="item__heart-box"><img src="" alt="Amei">
          <p class="item__heart-text">Amei</p>
        </div>
        <div class="item__bag-box"><img src="" alt="Comprar">
          <p class="item__bag-text">Comprar</p>
        </div>
      </div>
    </div>
  </section>
  <section class="item">
    <figure class="item__photo-container"><img class="item__photo" src="" alt="Photo"></figure>
    <div class="item__texts">
      <p class="item__title">Fórmula de lançamento</p>
      <p class="item__author">Por <strong>Érico Rocha</strong></p>
      <p class="item__description"> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. </p>
    </div>
    <div class="item__buy">
      <div class="item__star-button">
        <p class="item__star-note">3.5</p><img class="item__star" src="" alt="estrela">
      </div>
      <div class="item__buttons">
        <p class="item__price">R$ 2550</p>
        <div class="item__heart-box"><img src="" alt="Amei">
          <p class="item__heart-text">Amei</p>
        </div>
        <div class="item__bag-box"><img src="" alt="Comprar">
          <p class="item__bag-text">Comprar</p>
        </div>
      </div>
    </div>
  </section>
  <section class="item">
    <figure class="item__photo-container"><img class="item__photo" src="" alt="Photo"></figure>
    <div class="item__texts">
      <p class="item__title">Fórmula de lançamento</p>
      <p class="item__author">Por <strong>Érico Rocha</strong></p>
      <p class="item__description"> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. </p>
    </div>
    <div class="item__buy">
      <div class="item__star-button">
        <p class="item__star-note">3.5</p><img class="item__star" src="" alt="estrela">
      </div>
      <div class="item__buttons">
        <p class="item__price">R$ 2550</p>
        <div class="item__heart-box"><img src="" alt="Amei">
          <p class="item__heart-text">Amei</p>
        </div>
        <div class="item__bag-box"><img src="" alt="Comprar">
          <p class="item__bag-text">Comprar</p>
        </div>
      </div>
    </div>
  </section>
  <section class="item">
    <figure class="item__photo-container"><img class="item__photo" src="" alt="Photo"></figure>
    <div class="item__texts">
      <p class="item__title">Fórmula de lançamento</p>
      <p class="item__author">Por <strong>Érico Rocha</strong></p>
      <p class="item__description"> Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. </p>
    </div>
    <div class="item__buy">
      <div class="item__star-button">
        <p class="item__star-note">3.5</p><img class="item__star" src="" alt="estrela">
      </div>
      <div class="item__buttons">
        <p class="item__price">R$ 2550</p>
        <div class="item__heart-box"><img src="" alt="Amei">
          <p class="item__heart-text">Amei</p>
        </div>
        <div class="item__bag-box"><img src="" alt="Comprar">
          <p class="item__bag-text">Comprar</p>
        </div>
      </div>
    </div>
  </section>
</section>
`;
